utilities/src/Types/Transaction.res

```diff
@@ -0,0 +1,145 @@
+}
+/**
+ * Data pulled from a `sentry-trace` header
+ */
+type traceparentData = Js.Json.t
+
+type transactionContext = {
+  ...Span.spanContext,
+  /**
+   * If true, sets the end timestamp of the transaction to the highest timestamp of child spans, trimming
+   * the duration of the transaction. This is useful to discard extra time in the transaction that is not
+   * accounted for in child spans, like what happens in the idle transaction Tracing integration, where we finish the
+   * transaction after a given "idle time" and we don't want this "idle time" to be part of the transaction.
+   */
+  trimEnd?: bool,
+  /**
+   * If this transaction has a parent, the parent's sampling decision
+   */
+  parentSampled?: bool,
+  /**
+   * Metadata associated with the transaction, for internal SDK use.
+   */
+  metadata?: option<transactionMetadata>,
+}
+
+type samplingContext = {
+  ...customSamplingContext,
+  /**
+   * Context data with which transaction being sampled was created
+   */
+  transactionContext: transactionContext,
+  /**
+   * Sampling decision from the parent transaction, if any.
+   */
+  parentSampled?: bool,
+  /**
+   * Object representing the URL of the current page or worker script. Passed by default when using the `BrowserTracing`
+   * integration.
+   */
+  location?: Misc.workerLocation,
+  /**
+   * Object representing the incoming request to a node server. Passed by default when using the TracingHandler.
+   */
+  request?: Misc.extractedNodeRequestData,
+}
+
+type transaction = {
+  ///TODO figure out how to exculde type while type spreading
+
+  //...transactionContext,// exculde <Span, 'setName' | 'name'>
+
+  /**
+   * @inheritDoc
+   */
+  spanId: string,
+  /**
+   * @inheritDoc
+   */
+  traceId: string,
+  /**
+   * @inheritDoc
+   */
+  startTimestamp: float,
+  /**
+   * @inheritDoc
+   */
+  tags: Js.Json.t,
+  /**
+   * @inheritDoc
+   */
+  data: Js.Json.t,
+  /**
+   * Metadata about the transaction
+   */
+  metadata: transactionMetadata,
+  /**
+   * The instrumenter that created this transaction.
+   */
+  instrumenter: Instrumenter.instrumenter,
+  /**
+   * Set the name of the transaction
+   */
+  setName: (~name: string, ~source: transactionSource=?, unit) => unit,
+  /**
+   * Set the context of a transaction event
+   */
+  setContext: (~key: string, ~context: Context.context) => unit,
+  /**
+   * Set observed measurement for this transaction.
+   *
+   * @param name Name of the measurement
+   * @param value Value of the measurement
+   * @param unit Unit of the measurement. (Defaults to an empty string)
+   */
+  setMeasurement: (~name: string, ~value: float, ~unit: Measurement.measurementUnit) => unit,
+  /** Returns the current transaction properties as a `TransactionContext` */
+  toContext: unit => transactionContext,
+  /** Updates the current transaction with a new `TransactionContext` */
+  updateWithContext: (~transactionContext: transactionContext) => unit,
```

Comment:
A function or callback taking a single param doesn't need to be named param,
it can just be 
`updateWithContext: (transactionContext) => unit,`
or
`updateWithContext: (transactionContext: transactionContext) => unit`

Reply:
Okay cool